Source/WTF/wtf/threads/Signals.cpp

```diff
@@ -203,6 +204,162 @@ kern_return_t catch_mach_exception_raise_state_identity(mach_port_t, mach_port_t
     return KERN_FAILURE;
 }
 
+static kern_return_t runSignalHandlers(Signal &signal, PlatformRegisters& registers, bool &didHandle, mach_msg_type_number_t dataCount, mach_exception_data_t exceptionData)
+{
+    SigInfo info;
+    SignalHandlers& handlers = g_wtfConfig.signalHandlers;
+    if (signal == Signal::AccessFault) {
+        ASSERT_UNUSED(dataCount, dataCount == 2);
+        info.faultingAddress = reinterpret_cast<void*>(exceptionData[1]);
+#if CPU(ADDRESS64)
+        // If the faulting address is out of the range of any valid memory, we would
+        // not have any reason to handle it. Just let the default handler take care of it.
+        static constexpr unsigned validAddressBits = OS_CONSTANT(EFFECTIVE_ADDRESS_WIDTH);
+        static constexpr uintptr_t invalidAddressMask = ~((1ull << validAddressBits) - 1);
+        if (bitwise_cast<uintptr_t>(info.faultingAddress) & invalidAddressMask)
+            return KERN_FAILURE;
+#endif
+    }
+
+    handlers.forEachHandler(signal, [&] (const SignalHandler& handler) {
+        SignalAction handlerResult = handler(signal, info, registers);
+        didHandle |= handlerResult == SignalAction::Handled;
+    });
+    return KERN_SUCCESS;
+}
+
+#ifdef EXCEPTION_IDENTITY_PROTECTED
+
+static thread_act_t threadIDGetThreadPort(uint64_t threadID)
+{
+    mach_msg_type_number_t threadCount;
+    thread_act_array_t threads;
+    struct thread_identifier_info threadIdentifierInfo;
+    unsigned infoCount = THREAD_IDENTIFIER_INFO_COUNT;
+    thread_act_t thread = THREAD_NULL;
+
+    task_threads(mach_task_self(), &threads, &threadCount);
+    for (mach_msg_type_number_t i = 0; i < threadCount; i++) {
+        if (thread != THREAD_NULL) {
+            // clean up all the unnecessary thread ports like a good citizen
+            ASSERT(thread != threads[i]);
+            mach_port_deallocate(mach_task_self(), threads[i]);
+            continue;
+        }
+
+        kern_return_t kr = thread_info(threads[i], THREAD_IDENTIFIER_INFO,
+        reinterpret_cast<thread_info_t>(&threadIdentifierInfo), &infoCount);
```

Comment:
Please indent this by 4 spaces.  Alternatively, just keep as part of the previous line, and don't have to break it out.